docs/contracts.rst

```diff
@@ -953,6 +953,31 @@ not known in the context of the class where it is used,
 although its type is known. This is similar for ordinary
 virtual method lookup.
 
+.. index:: ! constructor
+
+Constructors
+============
+Constructor is an optional function with the same name as the contract which is executed upon contract creation. 
+Constructor functions can be either ``public`` or ``internal``.
+
+::
+
+    pragma solidity ^0.4.19;
+
+    contract A {
+        uint public a;
+
+        function A(uint _a) internal {
+            a = _a;
+        }
+    }
+
+    contract B is A(1) {
+        function B() public {}
+    }
+
+A constructor set as ``internal`` causes the contract to be marked as :ref:`abstract <abstract-contract>`. Child contracts of an abstract contract can call the super constructor.
```

Comment:
Perhaps I'd extend this to say that when extending a contract the parent constructor is called implicitly an as such it is possible that a child contract inheriting an `internal` constructor from the parent makes it non-abstract. (Would need to phrase this in a more readable form.)

Then as a second feature to that of course one can call the super constructor even if it is marked `internal`.

Reply:
Since this is specific to the description of an abstract contracts I think I'll add it to that section if you don't mind.

Reply:
@axic I don't get what you mean here:
>child contract inheriting an internal constructor from the parent makes it non-abstract
A child contract inherits an abstract contract with internal constructor does not become non-abstract necessarily as if for instance it doesn't implement all of the abstract functions, it will be abstract.